src/ui/public/accessibility/__tests__/kbn_accessible_click.js

```diff
@@ -0,0 +1,120 @@
+    });
+
+    it(`when the element doesn't have an ng-click`, () => {
+      const html = `<div kbn-accessible-click></div>`;
+      expect(() => {
+        $compile(html)($rootScope);
+      }).to.throwError(`kbnAccessibleClick requires ng-click to be defined on its element.`);
+    });
+  });
+
+  describe(`doesn't throw an error`, () => {
+    it('when the element is a link without an href', () => {
+      const html = `<a ng-click="noop" kbn-accessible-click></a>`;
+      expect(() => {
+        $compile(html)($rootScope);
+      }).not.to.throwError(`kbnAccessibleClick doesn't need to be used on a link if it has a href attribute.`);
+    });
+  });
+
+  describe('adds accessibility attributes', () => {
+    it('tabindex', () => {
+      const html = `<div ng-click="noop" kbn-accessible-click></div>`;
+      const element = $compile(html)($rootScope);
+      expect(element.attr('tabindex')).to.be('0');
+    });
+
+    it('role', () => {
+      const html = `<div ng-click="noop" kbn-accessible-click></div>`;
+      const element = $compile(html)($rootScope);
+      expect(element.attr('role')).to.be('button');
+    });
+  });
+
+  describe(`doesn't override pre-existing accessibility attributes`, () => {
+    it('tabindex', () => {
+      const html = `<div ng-click="noop" kbn-accessible-click tabindex="1"></div>`;
+      const element = $compile(html)($rootScope);
+      expect(element.attr('tabindex')).to.be('1');
+    });
+
+    it('role', () => {
+      const html = `<div ng-click="noop" kbn-accessible-click role="submit"></div>`;
+      const element = $compile(html)($rootScope);
+      expect(element.attr('role')).to.be('submit');
+    });
+  });
+
+  describe(`calls ng-click`, () => {
+    let scope;
+    let element;
+
+    beforeEach(function () {
+      scope = $rootScope.$new();
+      scope.handleClick = sinon.stub();
+      const html = `<div ng-click="handleClick()" kbn-accessible-click></div>`;
+      element = $compile(html)(scope);
+    });
+
+    it(`on ENTER keyup`, () => {
+      const e = angular.element.Event('keyup'); // eslint-disable-line new-cap
+      e.keyCode = ENTER_KEY;
+      element.trigger(e);
+      sinon.assert.calledOnce(scope.handleClick);
+    });
+
+    it(`on SPACE keyup`, () => {
+      const e = angular.element.Event('keyup'); // eslint-disable-line new-cap
+      e.keyCode = SPACE_KEY;
+      element.trigger(e);
+      sinon.assert.calledOnce(scope.handleClick);
+    });
+  });
+
+  it(`doesn't call ng-click when the originating element is a child`, () => {
+    const scope = $rootScope.$new();
+    scope.handleClick = sinon.stub();
+    const html = `<div ng-click="handleClick()" kbn-accessible-click></div>`;
+    const element = $compile(html)(scope);
+    const child = angular.element(`<button></button>`);
+    element.append(child);
+    const e = angular.element.Event('keydown'); // eslint-disable-line new-cap
+    e.which = ENTER_KEY;
```

Comment:
Why are you setting `e.which` here when you were setting `e.keyCode` in the tests above?